kedro-telemetry/kedro_telemetry/plugin.py

```diff
@@ -16,20 +16,58 @@
 import yaml
 from kedro.framework.cli.cli import KedroCLI
 from kedro.framework.cli.hooks import cli_hook_impl
+from kedro.framework.hooks import hook_impl
 from kedro.framework.startup import ProjectMetadata
+from kedro.io.data_catalog import DataCatalog
+from kedro.pipeline import Pipeline
 
-from kedro_telemetry import __version__ as telemetry_version
+from kedro_telemetry import __version__ as TELEMETRY_VERSION
 from kedro_telemetry.masking import _get_cli_structure, _mask_kedro_cli
 
 HEAP_APPID_PROD = "2388822444"
-
 HEAP_ENDPOINT = "https://heapanalytics.com/api/track"
 HEAP_HEADERS = {"Content-Type": "application/json"}
 TIMESTAMP_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
 
 logger = logging.getLogger(__name__)
 
 
+def _hash(string: str) -> str:
+    return hashlib.sha512(bytes(string, encoding="utf8")).hexdigest()
```

Comment:
minor refactoring since this is called a couple of times and they all look the same.